tests/std/include/test_regex_support.hpp

```diff
@@ -116,6 +116,59 @@ class regex_fixture {
         }
     }
 
+    void should_capture_groups(const std::string& subject, const std::string& pattern,
+        const std::vector<std::string>& groups,
+        const std::regex_constants::syntax_option_type flags = std::regex_constants::ECMAScript) {
+        try {
+            const std::regex r(pattern, flags);
+            std::smatch m;
+
+            if (!std::regex_match(subject, m, r)) {
+                printf(R"(Expected regex("%s") to match "%s".)"
+                       "\n",
+                    pattern.c_str(), subject.c_str());
+                fail_regex();
+                return;
+            }
+
+            if (m[0] != subject) {
+                printf(R"(should_capture("%s", "%s"): not a full match)"

```

Comment:
(Occurs below) These messages are saying `should_capture`, but this function is `should_capture_groups` and takes additional arguments. If we fail, I think all of them should print out the groups, and also the value of `flags` as an `int`.